src/WebJobs.Script.WebHost/Security/KeyManagement/BaseSecretsRepository.cs

```diff
@@ -50,6 +50,8 @@ public BaseSecretsRepository(string secretsSentinelFilePath, ILogger logger, IEn
 
         public abstract bool IsEncryptionSupported { get; }
 
+        public abstract string Name { get; }
```

Comment:
Introducing a new property here to enable the provider to define this and avoid having to log/resolve everything based on the type name only (also avoid the calls to `GetType`  that would be required)